image_writers_guide/guidelines.adoc

```diff
@@ -5,143 +5,186 @@
+Docker reads the [sysitem]#Dockerfile# and runs the instructions from top to bottom. Every successfully executed instruction creates a layer which can be reused the next time this or another image is built. It is very important to place instructions that will rarely change at the top of your [sysitem]#Dockerfile#. Doing so ensures the next builds of the same image are very fast because the cache is not invalidated by upper layer changes.
 
-In addition, performing multiple commands in a single `RUN` statement will reduce the number of layers in your image which will improve download and extraction time.
+For example, if you are working on a [sysitem]#Dockerfile# that contains an `ADD` command to install a file you are iterating on, and a `RUN` command to `yum install` a package, it is best to put the `ADD` command last:
 
+----
+FROM foo
+RUN yum -y install mypackage && yum clean all -y
+ADD myfile /test/myfile
+----
 
-=== Place instructions in proper order
-Docker reads the Dockerfile and runs the instructions from top to bottom. Every successfully executed instruction creates a layer which can be reused next we build this (or other image). It is very important to place on top of your Dockerfile instructions that will change rarely.
+This way each time you edit `myfile` and rerun `docker build`, the system reuses the cached layer for the `yum` command and only generates the new layer for the `ADD` operation.
 
-This will make sure the next builds of the same image will be very fast because the cache will not be invalidated by upper layer changes.
+If instead you wrote the [sysitem]#Dockerfile# as:
 
-For example, if you are working on a Dockerfile that contains an `ADD` command to install a file you are iteratin on, and a `RUN` command to yum install a package, it's best to put the `ADD` command last, such as:
+----
+FROM foo
+ADD myfile /test/myfile
+RUN yum -y install mypackage && yum clean all -y
+----
 
-		FROM foo
-		RUN yum -y install mypackage && yum clean all -y
-		ADD myfile /test/myfile
+Then each time you changed `myfile` and reran `docker build`, the `ADD` operation would invalidate the `RUN` layer cache, so the `yum` operation would need to be rerun as well.
 
-This way each time you edit `myfile` and rerun `docker build` the system will reuse the cached layer for the yum command and only have to generate the new layer for the `ADD` operation.
+*Using `EXPOSE` to Mark Important Ports*
 
-If instead you wrote the Dockerfile as:
+See the "Always `EXPOSE` Important Ports" section of the http://www.projectatomic.io/docs/docker-image-author-guidance[Project Atomic documentation] for more information.
 
-		FROM foo
-		ADD myfile /test/myfile
-		RUN yum -y install mypackage && yum clean all -y
+*Setting Environment Variables with `ENV`*
```

Comment:
Can we use just use "Setting Environment Variables" in the heading? We start off by saying that you do it with 'ENV` anyway.